### ISSUE TITLE
Update TypeScript Defs to version 5.0.0

### ISSUE THREAD

**Issue by simonepri** [restify/node-restify#1389]
Restify 5.0.0 is finally released 🎉  so TypeScript definitions needs to be updated accordingly.
https://www.npmjs.com/package/@types/restify

**Comment by retrohacker** [comment]
Hey @simonepri!

Thanks for taking the time to open this issue! I wasn't aware that repository exists, it appears to be maintained by the DefinitelyTyped team. While we don't (currently) maintain TypeScript definitions for restify, it's really awesome to see community efforts like this!

I believe their repo is the best place for this request: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/new

**Comment by eddieajau** [comment]
@retrohacker would you be open to accepting a PR to include the typings with the restify package itself (I'm happy to do it)? It doesn't cost you anything and doesn't affect anyone using plain old JavaScript but getting anything changed via the DefinitelyTyped repository is an absolute nightmare. It would be a HUGE help to those of us using TypeScript.

Thanks in advance.

**Comment by simonepri** [comment]
@eddieajau I think that is still a better option to send a PR on DefinitelyTyped repo.
Currently there is an active discussion here:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17620

**Comment by retrohacker** [comment]
Hey @eddieajau,

To expand on @simonepri's answer, I'd like to work towards moving unrelated pieces of the restify codebase out into their own repos. The rationale for this is that we currently have quite a few components that _could_ live on their own as standalone projects (mostly in the plugin repo) yet we have bundled them up into the restify repo. While, in many cases, we had maintainers who were passionate about those plugins at one point, maintainers tend to be transient. When a maintainer moves on, the work for maintaining that plugin falls to other maintainers who may have no use for that contribution. This results in these code paths becoming neglected, giving both the maintainers and the community bad feels.

I'd like to work towards isolating the core restify codebase (which focuses on building semantically correct RESTful services optimized for introspection) from the components that could be separate (i.e. CORS and serving static content). Some of these repos may live under the restify org, while others can be moved out into the community.

I believe in this case, moving the module out into the community may be the best course. Over time, if there is enough interest and we have enough maintainers who use TypeScript bindings, we can talk about migrating the repo to the restify org. DefinitelyTyped seems to be the best place for this in the community, but we are happy to link to multiple repos in our documentation if you would like to maintain one while DefinitelyTyped gets updated :smile:

tl;dr: My order of preference is

1. Module maintained in userland
2. Module maintained under the restify org
3. Module living inside the node-restify repo

**Comment by eddieajau** [comment]
Thanks for the reply @retrohacker. My frustration with DT is that it breaks the S in SOLID. The DT repo is a monolith for every typings (hundreds of them) so while it's great for consumers, it's a terrible arrangement for producers (there is only one owner of the @types scope). My pitch to you is that if you allow typings to be added to each package, you might just get one or two of the TypeScript users to help with other areas of your code. It would also mean that the TypeScripters would be asking you lots of questions about things as they come up and we can solve them in your repo's rather than DT's.

As I said, it costs you nothing and I think you would find that having the typings would encourage contribution.

In terms of splitting your offering up, I'm happy to have a chat with you about it. My advice is to first split it up into your own org and then work out if you can farm off responsibility but still keep a modicum of control over the repo's. I've actually had to rollback to 4 because CORS has been dropped in 5, which is a shame.

**Comment by eddieajau** [comment]
How about this. What if I did a proof-of-concept for `restify-errors` and then you can see how you feel about it?

**Comment by simonepri** [comment]
@eddieajau 
CORS was not dropped, has been moved here:
https://github.com/TabDigital/restify-cors-middleware

I agree with @retrohacker. Your point of view isn't completely wrong but you should consider that the DefinitelyTyped repo is really active in terms of community and it guarantee the quality of the types.

**Comment by simonepri** [comment]
@eddieajau as you can see they are quite fast:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17765#issuecomment-313077551

**Comment by eddieajau** [comment]
Ok. Can't say I'm not disappointed but that's the way it goes :)
